app/assets/javascripts/admin/order_cycles/controllers/order_cycle_exchanges_controller.js.coffee

```diff
@@ -37,5 +37,12 @@ angular.module('admin.orderCycles')
       $event.preventDefault()
       OrderCycle.addDistributor($scope.new_distributor_id)
 
+    $scope.setPickupTimeFieldDirty = (index) ->
+      setTimeout ->
```

Comment:
I think in Angular it's recommended to use the built-in `$timeout` service instead of `window.setTimeout`. I think `setTimeout` can have issues in different browsers, maybe the `$timeout` service applies some workarounds?